main.rb

```diff
@@ -0,0 +1,181 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_map(&proc)
+    temp = []
+    (0..length - 1).each do |i|
+      temp << [proc.call(self[i])] if proc
+      temp << yield(self[i]) if block_given? && !proc
+    end
+    temp = to_enum if temp.empty?
+    temp
+  end
```

Comment:
* [ ] my_map returns a new array with the results of running block once for every element in enum.

Reply:
- Wrong method output